tests/integration/targets/aci_igmp_interface_policy/tasks/main.yml

```diff
@@ -0,0 +1,215 @@
+    - update_igmp_intf_policy.current.0.igmpIfPol.attributes.descr == "Updated IGMP Interface Policy"
+    - update_igmp_intf_policy.current.0.igmpIfPol.attributes.name == "ansible_igmp_intf_policy"
+    - update_igmp_intf_policy.current.0.igmpIfPol.attributes.grpTimeout == "250"
+    - update_igmp_intf_policy.current.0.igmpIfPol.attributes.queryIntvl == "150"
+    - update_igmp_intf_policy.current.0.igmpIfPol.attributes.rspIntvl == "7"
+    - update_igmp_intf_policy.current.0.igmpIfPol.attributes.lastMbrCnt == "4"
+    - update_igmp_intf_policy.current.0.igmpIfPol.attributes.lastMbrRespTime == "6"
+    - update_igmp_intf_policy.current.0.igmpIfPol.attributes.startQueryCnt == "4"
+    - update_igmp_intf_policy.current.0.igmpIfPol.attributes.startQueryIntvl == "6"
+    - update_igmp_intf_policy.current.0.igmpIfPol.attributes.querierTimeout == "180"
+    - update_igmp_intf_policy.current.0.igmpIfPol.attributes.robustFac == "5"
+    - update_igmp_intf_policy.current.0.igmpIfPol.attributes.ver == "v2"
+    - update_igmp_intf_policy.current.0.igmpIfPol.attributes.ifCtrl == "allow-v3-asm"
+
+- name: Verify an update without ifCtrl options present leaves existing ifCtrl in place
+  ansible.builtin.assert:
+    that:
+    - update_igmp_no_ifctrl is changed
+    - update_igmp_no_ifctrl.current.0.igmpIfPol.attributes.ver == "v3"
+    - update_igmp_no_ifctrl.current.0.igmpIfPol.attributes.ifCtrl == "allow-v3-asm"
+
+# QUERY IGMP INTERFACE POLICY
+- name: Query an IGMP Interface Policy
+  cisco.aci.aci_igmp_interface_policy:
+    <<: *aci_igmp
+    state: query
+  register: query_one
+
+- name: Query all IGMP Interface Policies
+  cisco.aci.aci_igmp_interface_policy:
+    <<: *aci_info
+    state: query
+  register: query_all
+
+- name: Verify IGMP Interface Policy queries
+  ansible.builtin.assert:
+    that:
+    - query_one is not changed
+    - query_one.current.0.igmpIfPol.attributes.dn == "uni/tn-ansible_tenant/igmpIfPol-ansible_igmp_intf_policy"
+    - query_one.current.0.igmpIfPol.attributes.descr == "Updated IGMP Interface Policy"
+    - query_one.current.0.igmpIfPol.attributes.name == "ansible_igmp_intf_policy"
+    - query_one.current.0.igmpIfPol.attributes.grpTimeout == "250"
+    - query_one.current.0.igmpIfPol.attributes.queryIntvl == "150"
+    - query_one.current.0.igmpIfPol.attributes.rspIntvl == "7"
+    - query_one.current.0.igmpIfPol.attributes.lastMbrCnt == "4"
+    - query_one.current.0.igmpIfPol.attributes.lastMbrRespTime == "6"
+    - query_one.current.0.igmpIfPol.attributes.startQueryCnt == "4"
+    - query_one.current.0.igmpIfPol.attributes.startQueryIntvl == "6"
+    - query_one.current.0.igmpIfPol.attributes.querierTimeout == "180"
+    - query_one.current.0.igmpIfPol.attributes.robustFac == "5"
+    - query_one.current.0.igmpIfPol.attributes.ver == "v3"
+    - query_one.current.0.igmpIfPol.attributes.ifCtrl == "allow-v3-asm"
+    - query_all is not changed
```

Comment:
Can you add multiple igmp interface policies and assert that query all is indeed returns more than 1 policy?

Reply:
test added